metadata/src/utils/ordered_map.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use std::collections::HashMap;
+
+/// A minimal ordered map to let one search for
+/// things by key or get the values in insert order.
+#[derive(Debug, Clone)]
+pub struct OrderedMap<K, V> {
```

Comment:
I'm not super happy about this because it's quite hard to understand such as `push_insert` which inserts one item at the end of the `Vec` and one item at `key` in the Map.

but fine :P

Reply:
I'm happy to hear any suggestions here! My goal is just to preserve the order of items as seen in metadata, but also allow access by some other key (name, mostly). It just exists so that there's no need to manually keep a vec and hashmap in sync for doing this, but perhaps there's a better approach or maybe some api changes that would make it easier to understand? :)

Reply:
ok, I didn't look at the code where it's used and yes I see why you added it